### PR TITLE
Add Ubuntu CodeBuild project - ARM

### PR DESCRIPTION
**Pull request by prateekchaudhry**
### Summary

This PR adds adds dev CodeBuild projects and to release CodePipeline for making ARM deb targets. A lot of changes are similar to a previous PR https://github.com/aws/amazon-ecs-agent/pull/3233 where we added CodeBuild projects to make AMD deb targets.

Similar to PR, adds Ubuntu CodeBuild project in Release CodePipeline stack to make release artifacts and sign appropriately, and modify the CF appropriately to transfer the artifacts between stages.

Some small changes and differences discussed in implementation details:

### Implementation details

* CodeBuild does not maintain an image for Ubuntu - ARM. Changes image to scratch Ubuntu 20.04 docker image from ECR ```public.ecr.aws/lts/ubuntu:20.04``` (https://gallery.ecr.aws/lts/ubuntu)
* Some changes in buildspec.
    * sudo removed because default Ubuntu docker image does not have user and logs into root by default
    * apt moved to apt-get - better suited for scripts
    * ```DEBIAN_FRONTEND=noninteractive``` set for making the ```apt install``` automated in script

### Testing

Tested by building the CFs and checking GitHub artifacts in S3 buckets

New tests cover the changes: No new tests
### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by fierlion — DISMISSED** [review]

File: build-infrastructure/codebuild-devbuild-stack.yml [hunk 1]
```diff
@@ -339,4 +393,57 @@ Resources:
...
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Resource:
+                  - !Join
+                    - ':'
+                    - - 'arn:aws:logs'
+                      - !Ref 'AWS::Region'
+                      - !Ref 'AWS::AccountId'
+                      - !Sub 'log-group:/aws/codebuild/${BuildProjectName}-ubuntu-arm'
```
Comment (YashdalfTheGray) [inline build-infrastructure/codebuild-devbuild-stack.yml:420]: Non-blocking; You can do this all in one `!Sub` command

```
!Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/codebuild/${BuildProjectName}-ubuntu-arm"
```

It also makes it a little bit easier to read, I think. There are a couple of other places where this applies as well.

File: build-infrastructure/release-pipeline-stack.yml [hunk 2]
```diff
@@ -285,6 +289,84 @@ Resources:
...
+    Properties:
+      Artifacts:
+        Type: CODEPIPELINE
+      ConcurrentBuildLimit: 10
+      Description: CodeBuild project running an Ubuntu docker image to build the ECS Agent deb packages
+      Environment:
+        ComputeType: BUILD_GENERAL1_SMALL
+        Image: 'public.ecr.aws/lts/ubuntu:20.04'
+        ImagePullCredentialsType: CODEBUILD
+        PrivilegedMode: true
```
Comment (YashdalfTheGray) [inline build-infrastructure/release-pipeline-stack.yml:355]: Question; do we need privileged mode here? I might be missing something but this is mostly for building Docker images within the Docker container, right?

File: buildspecs/merge-build-ubuntu.yml [hunk 3]
```diff
@@ -44,12 +40,14 @@ phases:
...
-      - sudo apt install -y debhelper
-      - sudo apt install -y golang
+      - DEBIAN_FRONTEND=noninteractive
+
+      - apt-get update -y
+      - apt-get install -y make
+      - apt-get install -y dpkg-dev
+      - apt-get install -y devscripts
+      - apt-get install -y debhelper
+      - apt-get install -y golang
```
Comment (YashdalfTheGray) [inline buildspecs/merge-build-ubuntu.yml:50]: Nit; can group these packages together in a single command

**Review by YashdalfTheGray — COMMENTED** [review]
Some little comments but nothing pressing 😃

**Review by YashdalfTheGray — APPROVED** [review]

File: build-infrastructure/release-pipeline-stack.yml (hunk 2 above)
Comment (prateekchaudhry) [reply build-infrastructure/release-pipeline-stack.yml:355]: Great catches, I don't think we need privileged mode, I'll check

**Review by fierlion — APPROVED** [review]

File: build-infrastructure/codebuild-devbuild-stack.yml [hunk 4]
```diff
@@ -34,7 +34,7 @@ Resources:
       Description: A CodeBuild project to build artifacts (AMD/x86_64). Builds are triggered by PR creation and updates, and artifacts are saved in S3
       Environment:
         ComputeType: BUILD_GENERAL1_SMALL
-        Image: 'aws/codebuild/standard:5.0'
+        Image: 'public.ecr.aws/lts/ubuntu:20.04'
```
Comment (YashdalfTheGray) [inline build-infrastructure/codebuild-devbuild-stack.yml:37]: Callout; using public ECR is 💯 👌

**Review by YashdalfTheGray — APPROVED** [review]

**Pull request merged**